src/test/scala/firrtlTests/InferReadWriteSpec.scala

```diff
@@ -0,0 +1,103 @@
+   * Neither the name of the Regents nor the names of its contributors
+     may be used to endorse or promote products derived from this
+     software without specific prior written permission.
+IN NO EVENT SHALL REGENTS BE LIABLE TO ANY PARTY FOR DIRECT, INDIRECT,
+SPECIAL, INCIDENTAL, OR CONSEQUENTIAL DAMAGES, INCLUDING LOST PROFITS,
+ARISING OUT OF THE USE OF THIS SOFTWARE AND ITS DOCUMENTATION, EVEN IF
+REGENTS HAS BEEN ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+REGENTS SPECIFICALLY DISCLAIMS ANY WARRANTIES, INCLUDING, BUT NOT
+LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+A PARTICULAR PURPOSE. THE SOFTWARE AND ACCOMPANYING DOCUMENTATION, IF
+ANY, PROVIDED HEREUNDER IS PROVIDED "AS IS". REGENTS HAS NO OBLIGATION
+TO PROVIDE MAINTENANCE, SUPPORT, UPDATES, ENHANCEMENTS, OR
+MODIFICATIONS.
+*/
+
+package firrtlTests
+
+class InferReadWriteSpec extends MiddleTransformSpec {
+  val transform = firrtl.passes.InlineInstances
+  "Infer ReadWrite Ports" should "infer readwrite ports" in {
+    val input = """
+circuit sram6t :
+  module sram6t :
+    input clk : Clock
+    input reset : UInt<1>
+    output io : {flip en : UInt<1>, flip wen : UInt<1>, flip waddr : UInt<8>, flip wdata : UInt<32>, flip raddr : UInt<8>, rdata : UInt<32>}
+
+    io is invalid
+    smem mem : UInt<32>[128]
+    node T_9 = eq(io.wen, UInt<1>("h00"))
+    node T_10 = and(io.en, T_9)
+    wire T_12 : UInt
+    T_12 is invalid
+    when T_10 :
+      T_12 <= io.raddr
+      skip
+    read mport T_13 = mem[T_12], clk
+    io.rdata <= T_13
+    node T_14 = and(io.en, io.wen)
+    when T_14 :
+      write mport T_15 = mem[io.waddr], clk
+      T_15 <= io.wdata
+      skip
+""".stripMargin
+
+    val check = """
+circuit sram6t :
+  module sram6t :
+    input clk : Clock
+    input reset : UInt<1>
+    output io : { flip en : UInt<1>, flip wen : UInt<1>, flip waddr : UInt<8>, flip wdata : UInt<32>, flip raddr : UInt<8>, rdata : UInt<32>}
+    skip
+    mem mem :
+      data-type => UInt<32>
+      depth => 128
+      read-latency => 1
+      write-latency => 1
+      readwriter => rw_0
+      read-under-write => undefined
+    node T_9 = eq(io.wen, UInt<1>("h0"))
+    node T_10 = and(io.en, T_9)
+    wire T_12 : UInt<8>
+    skip
+    node GEN_0 = validif(T_10, io.raddr)
+    node GEN_1 = mux(T_10, UInt<1>("h1"), UInt<1>("h0"))
+    skip
+    skip
+    node T_14 = and(io.en, io.wen)
+    node GEN_2 = validif(T_14, io.waddr)
+    node GEN_3 = validif(T_14, clk)
+    node GEN_4 = mux(T_14, UInt<1>("h1"), UInt<1>("h0"))
+    node GEN_5 = validif(T_14, io.wdata)
+    node GEN_6 = mux(T_14, UInt<1>("h1"), UInt<1>("h0"))
+    io.rdata <= mem.rw_0.rdata
+    skip
+    skip
+    skip
+    skip
+    skip
+    mem.rw_0.wmode <= GEN_4
+    mem.rw_0.data <= GEN_5
+    mem.rw_0.mask <= GEN_6
+    T_12 <= GEN_0
+    mem.rw_0.clk <= clk
+    mem.rw_0.en <= or(GEN_1, GEN_4)
+    mem.rw_0.addr <= mux(GEN_4, GEN_2, T_12)
+""".stripMargin
```

Comment:
This check is impossibly brittle, at the bare minimum we will probably change the number of skips emitted let alone the names of generated nodes.

A better way to check would be to look at the actual DefMemory in the Firrtl tree and check that it has the parameters desired.